pandas/core/ops.py

```diff
@@ -1799,14 +1799,33 @@ def to_series(right):
             right = to_series(right)
 
         elif right.ndim == 2:
-            if left.shape != right.shape:
+            if right.shape == left.shape:
+                right = left._constructor(right, index=left.index,
+                                          columns=left.columns)
+
+            elif right.shape[0] == left.shape[0] and right.shape[1] == 1:
+                # Broadcast across columns
+                try:
+                    right = np.broadcast_to(right, left.shape)
+                except AttributeError:
+                    # numpy < 1.10.0
+                    right = np.tile(right, (1, left.shape[1]))
+
+                right = left._constructor(right,
+                                          index=left.index,
+                                          columns=left.columns)
+                # TODO: Double-check this doesn't make copies
```

Comment:
is this relevant?

Reply:
For performance, if the answer is that it does make copies, then yes.  At least in the sufficiently-new numpy case, we're passing a view in `left._constructor`.

```
a = np.arange(3)
b = a.reshape(3, 1)
c = np.broadcast_to(b, (3, 2))
d = c.copy()

df = pd.DataFrame(c)
df2 = pd.DataFrame(d)

>>> df.values.base is a  # <-- the concern is that this comes back False
True

>>> df2.values.base is d
True
```

In this example its OK.  I left the comment to do a more thorough check.  Are you confident this is always OK?